src/routes/chunk/handlers.ts

```diff
@@ -65,39 +68,117 @@ export const createChunkOffsetHandler = ({
 
       span.setAttribute('chunk.absolute_offset', offset);
 
-      // TODO: use a binary search to get this from the chain if it's not
-      // available in the DB
-      const {
-        data_root,
-        id,
-        data_size,
-        offset: weaveOffset,
-      } = await db.getTxByOffset(offset);
+      // Try to get transaction info from database first
+      let dbResult;
```

Comment:
Let's extract the fallback logic into a `CompositeTxOffsetSource` that implements a new `TxOffsetSource` interface (assuming we don't already have an equivalent interface). We should instantiate it in `system.ts` with the DB as the first source and the arweave composite client as the fallback source.